graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/views/ElasticsearchBackend.java

```diff
@@ -84,109 +81,91 @@ public class ElasticsearchBackend implements QueryBackend<ESGeneratedQueryContex
     public ElasticsearchBackend(Map<String, Provider<ESSearchTypeHandler<? extends SearchType>>> elasticsearchSearchTypeHandlers,
                                 JestClient jestClient,
                                 IndexLookup indexLookup,
-                                QueryStringDecorators queryStringDecorators,
                                 ESGeneratedQueryContext.Factory queryContextFactory,
                                 UsedSearchFiltersToQueryStringsMapper usedSearchFiltersToQueryStringsMapper,
-                                @Named("allow_leading_wildcard_searches") boolean allowLeadingWildcard,
-                                ObjectMapper objectMapper) {
+                                @Named("allow_leading_wildcard_searches") boolean allowLeadingWildcard) {
         this.elasticsearchSearchTypeHandlers = elasticsearchSearchTypeHandlers;
         this.jestClient = jestClient;
         this.indexLookup = indexLookup;
 
-        this.queryStringDecorators = queryStringDecorators;
         this.queryContextFactory = queryContextFactory;
         this.usedSearchFiltersToQueryStringsMapper = usedSearchFiltersToQueryStringsMapper;
         this.allowLeadingWildcard = allowLeadingWildcard;
     }
 
-    private QueryBuilder normalizeQueryString(String queryString) {
+    private QueryBuilder translateQueryString(String queryString) {
         return (queryString.isEmpty() || queryString.trim().equals("*"))
                 ? QueryBuilders.matchAllQuery()
                 : QueryBuilders.queryStringQuery(queryString).allowLeadingWildcard(allowLeadingWildcard);
     }
 
     @Override
-    public ESGeneratedQueryContext generate(SearchJob job, Query query, SearchConfig searchConfig) {
+    public ESGeneratedQueryContext generate(SearchJob job, Query query, Set<SearchError> validationErrors) {
         final BackendQuery backendQuery = query.query();
 
-        validateQueryTimeRange(query, searchConfig);
-
         final Set<SearchType> searchTypes = query.searchTypes();
 
-        final String queryString = this.queryStringDecorators.decorate(backendQuery.queryString(), job, query);
-        final QueryBuilder normalizedRootQuery = normalizeQueryString(queryString);
+        final QueryBuilder normalizedRootQuery = translateQueryString(backendQuery.queryString());
 
         final BoolQueryBuilder boolQuery = QueryBuilders.boolQuery()
                 .filter(normalizedRootQuery);
 
         usedSearchFiltersToQueryStringsMapper.map(query.filters())
                 .stream()
-                .map(searchFilterQueryString -> this.queryStringDecorators.decorate(searchFilterQueryString, job, query))
-                .map(this::normalizeQueryString)
+                .map(this::translateQueryString)
                 .forEach(boolQuery::filter);
 
         // add the optional root query filters
         generateFilterClause(query.filter(), job, query)
```

Comment:
"job" is now not needed in generateFilterClause() method and can be removed from the parameter list.

Reply:
So is "query" parameter, so probably "filter" is the only parameter that is needed after "decoration" removal.

Reply:
I have already prepared a follow-up PR cleaning up our search job passing, I did not want to make this PR even bigger.